src/Compilers/Core/Portable/SpecialTypeExtensions.cs

```diff
@@ -57,6 +57,35 @@ public static bool IsBlittable(this SpecialType specialType)
             }
         }
 
+        /// <summary>
+        /// Checks if a type is an immutable value type primitive.
+        /// </summary>
+        public static bool IsImmutableValueTypePrimitive(this SpecialType specialType)
```

Comment:
Do you really need this method? If it is a value type and has any SpecialType, then it should be safe. No?